src/main/java/org/jboss/windup/operator/util/WindupDeployment.java

```diff
@@ -0,0 +1,596 @@
+
+  KubernetesClient k8sClient;
+
+  private WindupResource windupResource;
+
+  private String deployment_executor;
+  private String deployment_postgre;
+  private String volume_postgresql_claim;
+  private String volume_mtaweb_claim;
+  private String volume_postgresql;
+  private String volume_executor;
+  private String volume_mtaweb;
+  private String volume_mtaweb_data;
+
+  private String mq_cluster_password;
+  private String db_username;
+  private String db_password;
+  private String sso_secret;
+  private String jgroups_cluster_password;
+
+  private String application_name;
+
+  private String deployment_amq;
+
+  private String namespace;
+
+  private String serviceAccount;
+
+  public WindupDeployment(WindupResource windupResource, MixedOperation<WindupResource, WindupResourceList, WindupResourceDoneable, Resource<WindupResource, WindupResourceDoneable>> crClient, 
+                          KubernetesClient k8sClient, String namespace, String serviceAccount) {
+    this.windupResource = windupResource;
+    this.crClient = crClient;
+    this.k8sClient = k8sClient;
+    this.namespace = namespace;
+    this.serviceAccount = serviceAccount;
+    initParams();
+  }
+
+  private void initParams() {
+    // Init names of objects
+    application_name = windupResource.getMetadata().getName();
+    volume_postgresql_claim = application_name + "-postgresql-claim";
+    deployment_postgre = application_name + "-postgresql";
+    deployment_executor = application_name + "-executor";
+    volume_mtaweb_claim = application_name + "-mta-web-claim";
+    volume_postgresql = application_name + "-postgresql-pvol";
+    volume_executor = application_name + "-mta-web-executor-volume";
+    volume_mtaweb = application_name + "-mta-web-pvol";
+    volume_mtaweb_data = application_name + "-mta-web-pvol-data";
+    deployment_amq = application_name + "-amq";
+
+
+    mq_cluster_password =  StringUtils.defaultIfBlank(windupResource.getSpec().getMq_cluster_password(),RandomStringUtils.randomAlphanumeric(8));
+    db_username =  StringUtils.defaultIfBlank(windupResource.getSpec().getDb_username(),"user" + RandomStringUtils.randomAlphanumeric(3));
+    db_password =  StringUtils.defaultIfBlank(windupResource.getSpec().getDb_password(),RandomStringUtils.randomAlphanumeric(8));
+    sso_secret = StringUtils.defaultIfBlank(windupResource.getSpec().getSso_secret(),RandomStringUtils.randomAlphanumeric(8));
+    jgroups_cluster_password  = StringUtils.defaultIfBlank(windupResource.getSpec().getJgroups_cluster_password(),RandomStringUtils.randomAlphanumeric(8));
+  }
+
+  public void deploy() {
+    // We are adding one by one instead of createOrReplace(volumes.toArray(new Volume[2])) 
+    // because in that case we receive an error : Too Many Items to Create
+    initCRStatusOnDeployment();
+
+    List<PersistentVolumeClaim> volumes = createVolumes();
+    k8sClient.persistentVolumeClaims().inNamespace(namespace).createOrReplace(volumes.get(0));
```

Comment:
Can we move to use something like a `forEach` to manage this and the following lists?

Reply:
:+1:  
I was using the index selection because of testing, now it doesnt make sense